docs/source/kinds_of_types.rst

```diff
@@ -356,6 +356,43 @@ check against ``None`` in the if condition.
 
     ``--strict-optional`` is experimental and still has known issues.
 
+.. _noreturn:
+
+The NoReturn type
+*****************
+
+Mypy also provides an experimental support for functions that supposed to
```

Comment:
It would be reasonable to discuss adding `NoReturn` to `typing` in the typing issue tracker. Features in `mypy_extensions` are non-standard, and they may be experimental but aren't necessarily.